Refactored Tratis/ThanksListWidget

DavidAvigdor1996 opened the pull request:
Combined the 2 lists into a single widget, as having both was redundent.
The refactor consisted using conditionals to identify the widget, instead of having 2 seperate files for it. 
Creating a util file with a function that given the identificier of the page, returns the needed information for it. 
And seperating a section (ShowAll) to a different file, in order to make the now-main file of this change (list_widgets)
In the process, fixed issue #95
Also, fixed a bug with a use of `DebugPrint` that prevented reseting the data.

Tovli commented on lib/MainPageHelpers/list_util.dart:
```diff
@@ -0,0 +1,24 @@
+    case PagesCode.GratitudeJournal:
+      return {
+        'mainTitle': locale!.homePageThanksMainTitle(gender),
+        'SecondaryTitle': locale!.homePageThanksSecondaryTitle(gender),
```
Secondary start with capital, main start with lowercase

Tovli reviewed (changes requested):
Partly reviewed, will continue tomorrow

Tovli commented on lib/MainPageHelpers/list_widgets.dart:
```diff
@@ -0,0 +1,483 @@
+// the trait list widget, it shows the list of the traits
+// this code is related to "מעלות" section in homepage.
+// this code is similar to thanksListWidget.dart .
+class ListWidget extends StatefulWidget {
```
I think a better decomposition would be to separate the presentation layer from the business layer
a 250 lines build function is a smell and you should have a very good reason to have such long function
see [this](https://github.com/AppFlowy-IO/AppFlowy/tree/main/frontend/appflowy_flutter/lib/features/shared_section) for example

DavidAvigdor1996 commented:
> I suggest you'll revert the refactoring - have only the two fixes and let's discuss some refactoring principles I think that reviewing [other flutter apps](https://github.com/Solido/awesome-flutter?tab=readme-ov-file#open-source-apps) will give you great value

The initial idea of this PR was to start the refactoring, but not do the entire thing in one go, as those are a lot of changes to go over, and I wanted to focus the pr on a simple and quick fix.
Now, I've finished the refactoring(as far as I know), addressing all of the issues that you have raise and then some.
I didn't want to reverse what I did already, because the state that the lists were in was really error-prone, with a lot of useless duplication.
Please call me up before/after you have gone over this, or generally when you have the time.

DavidAvigdor1996 commented on lib/MainPageHelpers/MainPageList/list_utils.dart:
```diff
@@ -0,0 +1,132 @@
+void addThankYou(String thankYou, UserInformation userInfoProvider,
+    stateFunction, popupFunction) {
+  List<String> thankYous_temp = userInfoProvider.thanks['thanks'] ?? [];
+  List<String> dates_temp = userInfoProvider.thanks['dates'] ?? [];
```
SharedPreferences cannot save date type, it can only save String and List<String>, so we have to default to that. I'll look if there is a way around this, this feature wasn't done originally by me and I didn't want to tinker too much with the implementation itself

DavidAvigdor1996 commented on lib/MainPageHelpers/MainPageList/list_utils.dart:
```diff
@@ -0,0 +1,132 @@
+  }
+}
+
+List<String> todayThankYousFunc(List<String> thankYous, List<String> dates) {
```
At some point, we need to filter the dates. I decided to let the function in this file handle all that logic. They get (thanks for ) ALL the dates and use/return the current date.

DavidAvigdor1996 commented on lib/MainPageHelpers/MainPageList/list_utils.dart:
```diff
@@ -0,0 +1,132 @@
+  return todayThankYous;
+}
+
+List<String> getListItems(PagesCode pageCode, UserInformation userInfoProvider,
```
Bit of organization and make the "view" file cleaner
